cmd/makedocs/main.go

```diff
@@ -59,15 +86,82 @@ func main() {
 	}
 }
 
-func getDesc(name string) (string, error) {
-	b, err := ioutil.ReadFile(checkersPath + name + ".md")
-	return string(b), err
+func parseComment(text string, c *checker) {
+	lines := strings.Split(text, "\n")
+	index := 0
+	stages := []func(l []string, i *int, c *checker) error{
+		parseShortDesc,
+		parseDesc,
+		parseBefore,
+		parseAfter,
+		parseNote,
+	}
+	for _, st := range stages {
+		err := st(lines, &index, c)
+		if err != nil {
+			log.Println(err)
+		}
+	}
+}
+
+func parseShortDesc(lines []string, index *int, c *checker) error {
+	c.ShortDescription = strings.TrimSpace(lines[0][1:]) + "\n\n"
+	*index += 2 // skip empty line
+	return nil
 }
 
-func shortDescription(desc string) string {
-	// We expect every checker to have first doc line
-	// to be a short summary, complete sentence.
-	summary := strings.Split(desc, "\n")[0]
-	summary = strings.TrimSpace(summary)
-	return summary
+func parseDesc(lines []string, index *int, c *checker) error {
+	if len(lines) <= *index {
+		return errors.New("parseDesc: no description provided")
+	}
+	if strings.HasPrefix(lines[*index], "@") { // if no description
+		return nil
+	}
+	for *index < len(lines) && len(lines[*index]) > 0 {
+		c.Description += strings.TrimSpace(lines[*index]) + "\n"
+		*index++
+	}
+	*index++ //skip empty line
```

Comment:
You could do every stage parsing function a method of some type, which would store index inside itself,
and do something like:
```go
p.skipEmptyLine()
```
Which would be more readable and can also perform a check that current like is really empty or not.

Reply:
Can be refactored later.